### ISSUE TITLE
Regression with `routes` option

### ISSUE THREAD

**Issue by sinedied** [BrowserSync/browser-sync#1064]
### Issue details
Creating instance with
```
var instance = browserSync.init({
    startPath: '/',
    server: {
          baseDir: baseDir,
          routes: null,
          middleware: []
   },
    browser: 'default'
  });
```
was working on 2.11.*, it crashes on start with 2.12.1.

Removing the `routes` option fixes the issue.

### Steps to reproduce/test case

See up

### Please specify which version of Browsersync, node and npm you're running

- Browsersync [  2.12.1  ]
- Node        [  5.0.0  ]
- Npm         [  3.8.2  ]

### Affected platforms

Tested on OSX and Windows

- [ ] linux
- [x] windows
- [x] OS X
- [ ] freebsd
- [ ] solaris
- [ ] other 

### Browsersync use-case

- [x] API
- [x] Gulp
- [ ] Grunt
- [ ] CLI

**Comment by shakyShane** [comment]
why is `routes` set to null?

**Comment by sinedied** [comment]
Because it may (or not) be overriden by user settings, and it was a fallback.
Using undefined instead of null solves this, but this simple change has broken many of our projects :(

**Comment by shakyShane** [comment]
@sinedied fair enough, as the behaviour has changed since 2.11 I will take a look at this

**Comment by gbrassey** [comment]
2.12 has also introduced an error when setting the routes property inside proxy settings
```
  proxy: {
    target: 'localhost:3000',
    middleware: [wpMiddleware, hotMiddleware],
    routes: {
      '/coverage': 'coverage',
    }
  },
```
I realize that this is not in the docs but it worked in 2.11 and we were using it to serve coverage reports next to our project in development.

**Comment by shakyShane** [comment]
@gbrassey can you log that as a separate issue with a small runnable example? :)

**Comment by shakyShane** [comment]
released at browser-sync@2.12.3

Thanks for reporting :)
